rand_core/src/os.rs

```diff
@@ -36,51 +36,53 @@ use getrandom::getrandom;
 ///
 /// # Usage example
 /// ```
-/// use rand_core::{RngCore, OsRng};
+/// use rand_core::{TryRngCore, OsRng};
 ///
 /// let mut key = [0u8; 16];
-/// OsRng.fill_bytes(&mut key);
-/// let random_u64 = OsRng.next_u64();
+/// OsRng.try_fill_bytes(&mut key).unwrap();
+/// let random_u64 = OsRng.try_next_u64().unwrap();
 /// ```
 ///
 /// [getrandom]: https://crates.io/crates/getrandom
 #[cfg_attr(doc_cfg, doc(cfg(feature = "getrandom")))]
 #[derive(Clone, Copy, Debug, Default)]
 pub struct OsRng;
```

Comment:
We *could* also rename this (we use `OsRng`, `entropy` and `getrandom`)?

Reply:
I think `OsRng` is a concise and good enough name. It may be a bit confusing in bare-metal contexts, i.e it's a bit weird to use "OS" RNG when you do not have any OS. We could name it something like `SysRng`, but I am not sure if it's worth the associated churn.

Somehow referencing `getrandom` would result in longer names and arguably will be less clear for users.